modules/client/src/internal/client/methods.ts

```diff
@@ -105,7 +105,7 @@ export class ClientMethods extends ClientCore implements IClientMethods {
       pluginInstallationData.push({
         pluginSetup: latestVersion[1],
         pluginSetupRepo: plugin.id,
-        data: plugin.data,
+        data: toUtf8String(plugin.data),
```

Comment:
Why are we passing a binary string here?

Reply:
because ethers doesn't convert the Uint8Array back to a string and ethers needs it as a string to work
That was why I initially changed it to be a string

Reply:
Oh, I wasn't aware of that. Ethers accepts many data types as binary data, doesn't it really work here?

Then I would suggest we lave your original solution, sorry for the confusion. Still, I assume the string should be a hex-encoded one, right? not a random binary string from the bytes

Reply:
let's leave it as it is.
it is an abi encoded hex string with a 0x prefix needed.

Reply:
In my local environment, calling `toUtf8String(new Uint8Array([1,2,3]))` returns `'\x01\x02\x03'` instead of `0x010203`.

Shouldn't it be our helper `bytesToHex()` as you initially added in the encoder?

Reply:
then it is this way. it works